apps/mobile/src/libs/components/checkbox-categories-form/checkbox-categories-form.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+	Button,
+	MultipleCheckboxInput,
+	View,
+} from "~/libs/components/components";
+import { NumericalValue } from "~/libs/enums/enums";
+import { useAppForm, useCallback, useEffect } from "~/libs/hooks/hooks";
+import { globalStyles } from "~/libs/styles/styles";
+import {
+	type CategoriesSelectedRequestDto,
+	type QuizScoresGetAllItemResponseDto,
+} from "~/libs/types/types";
+import { categoriesSelectedValidationSchema } from "~/packages/categories/categories";
+
+import {
+	CATEGORIES_FIELD_NAME,
+	CATEGORIES_FORM_DEFAULT_VALUES,
+	CHECK_ALL_CATEGORIES_NAME,
+} from "./libs/constants/constants";
+import {
+	getCategoriesSortedByScore,
+	getSelectedCategoryIds,
+} from "./libs/helpers/helpers";
+
+type Properties = {
+	categories: QuizScoresGetAllItemResponseDto[];
+	onSubmit: (payload: CategoriesSelectedRequestDto) => void;
+	submitButtonName: string;
+};
+
+const CheckboxCategoriesForm: React.FC<Properties> = ({
+	categories,
+	onSubmit,
+	submitButtonName,
+}: Properties): JSX.Element | null => {
+	const { control, errors, handleSubmit, reset } = useAppForm<{
+		[CATEGORIES_FIELD_NAME]: number[];
+	}>({
+		defaultValues: {
+			[CATEGORIES_FIELD_NAME]: CATEGORIES_FORM_DEFAULT_VALUES,
```

Comment:
Not sure that we need a constant for an empty array. Maybe leave it like this?

```suggestion
			[CATEGORIES_FIELD_NAME]: [],
```